src/providers/hana/qgshanaconnection.cpp

```diff
@@ -534,6 +536,17 @@ QVector<QgsHanaLayerProperty> QgsHanaConnection::getLayers(
   return list;
 }
 
+QVector<QgsHanaLayerProperty> QgsHanaConnection::getLayersFull(
+  const QString &schemaName,
+  bool allowGeometrylessTables,
+  bool userTablesOnly )
+{
+  QVector<QgsHanaLayerProperty> layers = getLayers( schemaName, allowGeometrylessTables, userTablesOnly );
+  for ( int i = 0; i < layers.size(); ++i )
+    readLayerInfo( layers[i] );
```

Comment:
```suggestion
  const QVector<QgsHanaLayerProperty> layers = getLayers( schemaName, allowGeometrylessTables, userTablesOnly );
  for ( const QgsHanaLayerProperty &layer : layers )
    readLayerInfo( layer );
```

Use iterators where possible

Reply:
The layers[i] object is changed in readLayerInfo. Hence, it cannot be declared as const.

Reply:
I see, can you add a comment here, that this will modify the layer?

Reply:
Done in e03ea84.